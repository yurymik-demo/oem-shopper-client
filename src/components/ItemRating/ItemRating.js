import React from "react";
import _ from "lodash";

const ItemRating = ({ rating, length = 5 }) => {
  if (!rating) return null;
  const filledStars = rating;
  const emptyStars = length - rating;

  const renderStars = (count, isFilled) => {
    const iconName = isFilled ? "star" : "star_border";
    const stars = [];
    while (count > 0) {
      stars.push(count);
      count--;
    }
    return stars.map(key => (
      <i key={key} className="material-icons">
        {iconName}
      </i>
    ));
  };

  return (
    <div>
      {renderStars(filledStars, true)}
      {renderStars(emptyStars, false)}
    </div>
  );
};

export default ItemRating;
