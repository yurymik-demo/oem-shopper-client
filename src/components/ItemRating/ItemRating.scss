.item-rating-container {
  a.rating-star-link {
    margin-right: 0 !important;
  }
}
