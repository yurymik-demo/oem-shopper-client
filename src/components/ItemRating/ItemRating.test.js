import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import ItemRating from "./ItemRating";

jest.mock("../../routes/home/home", () => {
  return () => "home_route_component";
});

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <ItemRating />
    </BrowserRouter>,
    div
  );
  expect(div).toMatchSnapshot();
  ReactDOM.unmountComponentAtNode(div);
});

it("renders 2 out of 5 stars solid crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <ItemRating rating="2" />
    </BrowserRouter>,
    div
  );
  expect(div).toMatchSnapshot();
  ReactDOM.unmountComponentAtNode(div);
});

it("renders 2 out of 10 stars solid crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <ItemRating rating="2" length="10" />
    </BrowserRouter>,
    div
  );
  expect(div).toMatchSnapshot();
  ReactDOM.unmountComponentAtNode(div);
});
