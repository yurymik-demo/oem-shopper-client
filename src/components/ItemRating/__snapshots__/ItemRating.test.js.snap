// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 2 out of 5 stars solid crashing 1`] = `
<div>
  <div
    class="item-rating-container"
  >
    <a
      class="rating-star-link"
      href="/"
    >
      <i
        class="material-icons tiny"
      >
        star
      </i>
    </a>
    <a
      class="rating-star-link"
      href="/"
    >
      <i
        class="material-icons tiny"
      >
        star
      </i>
    </a>
    <a
      class="rating-star-link"
      href="/"
    >
      <i
        class="material-icons tiny"
      >
        star_border
      </i>
    </a>
    <a
      class="rating-star-link"
      href="/"
    >
      <i
        class="material-icons tiny"
      >
        star_border
      </i>
    </a>
    <a
      class="rating-star-link"
      href="/"
    >
      <i
        class="material-icons tiny"
      >
        star_border
      </i>
    </a>
  </div>
</div>
`;

exports[`renders 2 out of 10 stars solid crashing 1`] = `
<div>
  <div
    class="item-rating-container"
  >
    <a
      class="rating-star-link"
      href="/"
    >
      <i
        class="material-icons tiny"
      >
        star
      </i>
    </a>
    <a
      class="rating-star-link"
      href="/"
    >
      <i
        class="material-icons tiny"
      >
        star
      </i>
    </a>
    <a
      class="rating-star-link"
      href="/"
    >
      <i
        class="material-icons tiny"
      >
        star_border
      </i>
    </a>
    <a
      class="rating-star-link"
      href="/"
    >
      <i
        class="material-icons tiny"
      >
        star_border
      </i>
    </a>
    <a
      class="rating-star-link"
      href="/"
    >
      <i
        class="material-icons tiny"
      >
        star_border
      </i>
    </a>
    <a
      class="rating-star-link"
      href="/"
    >
      <i
        class="material-icons tiny"
      >
        star_border
      </i>
    </a>
    <a
      class="rating-star-link"
      href="/"
    >
      <i
        class="material-icons tiny"
      >
        star_border
      </i>
    </a>
    <a
      class="rating-star-link"
      href="/"
    >
      <i
        class="material-icons tiny"
      >
        star_border
      </i>
    </a>
    <a
      class="rating-star-link"
      href="/"
    >
      <i
        class="material-icons tiny"
      >
        star_border
      </i>
    </a>
    <a
      class="rating-star-link"
      href="/"
    >
      <i
        class="material-icons tiny"
      >
        star_border
      </i>
    </a>
  </div>
</div>
`;

exports[`renders without crashing 1`] = `<div />`;
